security/pkg/nodeagent/caclient/providers/citadel/client_test.go

```diff
@@ -107,3 +115,99 @@ func TestCitadelClient(t *testing.T) {
 		}
 	}
 }
+
+type mockTokenCAServer struct {
```

Comment:
is this new test related to the "workload cert rotate" added in this PR?

Reply:
We will first test the token both empty and invalid. As for the server side it will examine both the cert and token. This unit test will test the token logic first

Reply:
so the answer is this test actually has nothing to do with the code added in this PR, and you plan to add the corresponding tests in a later PR?